core/src/main/java/com/linecorp/armeria/internal/common/DefaultRequestTarget.java

```diff
@@ -645,7 +654,9 @@ private static RequestTarget slowForClient(String reqTarget,
                                             null,
                                             -1,
                                             encodedPath,
-                                            encodedPath, encodedQuery,
+                                            encodedPath,
+                                            encodedPath,
```

Comment:
tbh I am not sure what's the best behavior for client, I just reuse the encodedPath

Reply:
I prefer using `reqTarget`, the original value we get from `:path` header.